src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1166,18 +1166,25 @@ void OrbitMainWindow::LoadCaptureOptionsIntoApp() {
   app_->SetTraceGpuSubmissions(settings.value(kTraceGpuSubmissionsSettingKey, true).toBool());
   app_->SetEnableApi(settings.value(kEnableApiSettingKey, true).toBool());
   app_->SetEnableIntrospection(settings.value(kEnableIntrospectionSettingKey, false).toBool());
-  bool const collect_callstack_on_thread_state_change =
-      settings
-          .value(kEnableCallStackCollectionOnThreadStateChanges,
-                 orbit_qt::CaptureOptionsDialog::kThreadStateChangeCallStackCollectionDefaultValue)
-          .toBool();
-  if (settings.value(kCollectThreadStatesSettingKey, false).toBool()) {
-    if (!collect_callstack_on_thread_state_change) {
-      app_->SetThreadStateChangeCallstackCollection(
-          CaptureOptions::kNoThreadStateChangeCallStackCollection);
+
+  if (absl::GetFlag(FLAGS_tracepoint_callstack_collection)) {
+    bool const collect_callstack_on_thread_state_change =
+        settings
+            .value(
+                kEnableCallStackCollectionOnThreadStateChanges,
+                orbit_qt::CaptureOptionsDialog::kThreadStateChangeCallStackCollectionDefaultValue)
+            .toBool();
+    if (settings.value(kCollectThreadStatesSettingKey, false).toBool()) {
+      if (!collect_callstack_on_thread_state_change) {
+        app_->SetThreadStateChangeCallstackCollection(
+            CaptureOptions::kNoThreadStateChangeCallStackCollection);
```

Comment:
Offtopic question: Why is this method not simply accepting a bool?

Reply:
Shouldn't we have an unspecified option? That's why we use an enum instead of a bool.

Reply:
Right. So this is for metrics collections. Fine for me.